test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -1737,4 +1743,47 @@ protected void skipTestWaitingForLuceneFix(org.apache.lucene.util.Version lucene
         fail("Remove call of skipTestWaitingForLuceneFix in " + RandomizedTest.getContext().getTargetMethod());
     }
 
+    protected void tcpReadinessProbeTrue(ReadinessService readinessService) throws Exception {
```

Comment:
This base class is already massive, can we avoid adding all these helpers here that are specific to readiness tests?

Reply:
Sounds good I'll move the code out into readiness classes.